code/modules/flufftext/Hallucination.dm

```diff
@@ -65,6 +65,9 @@ GLOBAL_LIST_INIT(hallucination_list, list(
 
 //Returns a random turf in a ring around the target mob, useful for sound hallucinations
 /datum/hallucination/proc/random_far_turf()
+	var/turf/target_T = get_turf(target)
```

Comment:
Because apparently target.z is 0 while they are not located on a turf, thanks Byond.